code/game/objects/items/devices/minedetector.dm

```diff
@@ -0,0 +1,102 @@
+/obj/item/device/mine_detector
+	name = "mine detector"
+	desc = "A device capable of detecting mines and traps in a range."
+
+	///The types that are detected by this device
+	var/list/detectable_types = list(/obj/item/landmine, /obj/item/storage/box/traps)
+
+	///The range in which the detection happens
+	var/detection_range = 4
+
+	///The angle from the center in which the detection happens
+	var/detection_angle = 30
+
+	///The last time this was activated, to avoid spamming it. In `world.time`
+	var/last_use = null
+
+	///How much to wait before allowing a new scan
+	var/delay_between_scans = 3 SECONDS
+
+	///If this detector is able to perform scans automatically, instead of having to trigger them manually
+	var/can_scan_automatically = FALSE
+
+	///The timer ID for the automatic rescans
+	var/timer_id = null
+
+/obj/item/device/mine_detector/attack_self(mob/user)
+	. = ..()
+
+	//Apply the delay and check it is passed, so a new scan can be done
+	if((last_use + delay_between_scans) > world.time)
+		return
+
+	last_use = world.time
+
+	if(!can_scan_automatically)
+		perform_scan(user)
+
+	else
+
+		//If there's no timer, start a new one, otherwise stop the callbacks (aka stop the automatic scanning)
+		if(!timer_id)
+			to_chat(user, SPAN_NOTICE("You power on the automatic mine detection system, which shows a loading screen with a progress bar."))
+			timer_id = addtimer(CALLBACK(src, PROC_REF(perform_scan), user), delay_between_scans, TIMER_UNIQUE|TIMER_STOPPABLE)
+		else
+			to_chat(user, SPAN_NOTICE("You power off the automatic mine detection system."))
+			deltimer(timer_id)
+			timer_id = null
+
+
+/obj/item/device/mine_detector/proc/perform_scan(mob/user)
+
+	var/list/turfs_scanned = get_turfs_in_cone(get_turf(src), dir2angle(user.dir), detection_range, detection_angle)
+
+	//This will contain the distance of the closest detected object
+	var/minimum_distance_detected = null
+
+	var/obj/effect/overlay/teleport_pulse/scanned = new() //This goes in nullspace because we want to project the image to the user only
+
+	//Search across the content of the turfs we got in the cone
+	for(var/turf/turf as anything in turfs_scanned)
+
+		//Show the effect on every turf that was scanned, to the user
+		user.client.images += image(scanned, turf)
```

Comment:
i vaguely remember making the second argument the turf instead of null attaches it to the turf instead of displaying it only to the user, could you test it with two clients connected to the same server to validate that this works as intended?

Reply:
This wouldn't really make sense, that's how you tell the engine where to show the image for the user, otherwise how would it know where to display it? I also don't have two clients

Reply:
the way i've done it in the past is doing

var/image/I = image(thing)
I.loc = turf
user.client.images = I

you can make another byond account, host the server on your main account by right clicking the top bar, clicking host, logging out, signing in with the new account, and then joining the localhosted server

Reply:
oh! for another example, you can take a look at the mining ore scanner, that also adds images to the user's client view

Reply:
As per https://secure.byond.com/docs/ref/info.html#/image

"One reason for creating images is player-by-player control over visibility. Images only become visible when they are explicitly output to players"